### PR TITLE
Extend documentation

### PR DESCRIPTION
**Pull request by RalfG**
This PR extends the [mzpaf.readthedocs.io](https://mzpaf.readthedocs.io) website with documentation for the LARK grammar, JSON Schema, regular expressions.

For some sections in the documentation, the Sphinx `conf.py` script automatically generates some markdown files from JSON files, and copies the railroad diagram. No GitHub actions are required for this, as this code runs on docs-generation at readthedocs.io.

The version of the documentation from this branch can be viewed on https://mzpaf.readthedocs.io/en/add-sphinx-docs/.

_Edit 11/11/2024: This PR was updated to add upon the existing documentation._

### CODE REVIEW

File: docs/implementations/python/api.rst [hunk 1]
```diff
@@ -7,6 +7,8 @@ Python API
     :imported-members:
 
 
+    .. manually documented as parse_annotation is undocumented
```
Comment (mobiusklein) [inline docs/implementations/python/api.rst:10]: I'll add a cross-reference to the manual documentation here:
```rst
:meth:`AnnotationStringParser.__call__`
```

The `autodoc` sees that it is a function and pulls the signature from `AnnotationStringParser.__call__`, but nothing else.

File: docs/_static/img/lark-railroad-diagram.svg
Comment (mobiusklein) [inline docs/_static/img/lark-railroad-diagram.svg:1]: Is this a copy of the generated SVG, or a symlink to it?

**Review by mobiusklein — APPROVED** [review]

**Review by edeutsch — APPROVED** [review]

File: docs/_static/img/lark-railroad-diagram.svg
Comment (RalfG) [reply docs/_static/img/lark-railroad-diagram.svg:1]: The file is copied over when the docs are generated: 

https://github.com/HUPO-PSI/mzPAF/blob/ec0f9cfefbbf5453bf8754374d140ab4c8132adf/docs/conf.py#L44-L48

It shouldn't necessarily be duplicated in the repo itself. I think I committed it by accident.

File: docs/implementations/python/api.rst (hunk 1 above)
Comment (RalfG) [reply docs/implementations/python/api.rst:10]: Not entirely following here, but feel free to update

File: docs/_static/img/lark-railroad-diagram.svg
Comment (mobiusklein) [reply docs/_static/img/lark-railroad-diagram.svg:1]: Thanks for pointing that out, I didn't make the connection there.

**Pull request merged**